lib/instrumentation/index.js

```diff
@@ -164,6 +167,13 @@ Instrumentation.prototype.start = function () {
     }
   }
 
+  const lambdaHandlerInfo = getLambdaHandler(process.env)
```

Comment:
It would be good to save this `lambdaHandlerInfo` on the Instrumentation object to not have to call `getLambdaHandler()` multiple times below.

Reply:
We could, but I don't like the idea of the Instrumentation object duplicating state that's available from the environment.  I also don't like the idea that we'll have an instrumentation object that will behave differently when that state's set vs. when it's not set.  

@trentm What's the goal/benefit of removing those function calls and replacing it with that data set on the instrumentation object?

Reply:
>  I also don't like the idea that we'll have an instrumentation object that will behave differently when that state's set vs. when it's not set.

It would always be set if you fetched it once in the constructor, or in Instrumentation#start.

> I don't like the idea of the Instrumentation object duplicating state that's available from the environment.

It feels equivalent (to me) to having `agent._conf` be an object that (at least partially) duplicates config state that was specified in the environment.

> What's the goal/benefit

My main concern was that the `getLambdaHandler()` processing is being done potentially many times:
- once in Instrumentation#start()
- and twice for every hooked module the way it is called in _startHook and _patchModule.

Reply:
Hmmm -- something about changing the base instrumentation object like this still bugs me but it's probably not worth tracking down right now.  I've added the `getLambdaHandlerInfo` and addPatch call to the constructor in cd54022a2ba42c1f3f0a71db30a88e68685594d5